runtime/sema/type.go

```diff
@@ -2332,6 +2329,12 @@ func (t *SpecialFunctionType) GetMembers() map[string]MemberResolver {
 	return withBuiltinMembers(t, members)
 }
 
+// ConstructorFunctionType is the the type representing a constructor function
+
+type ConstructorFunctionType struct {
+	*FunctionType
+}
```

Comment:
Could we use a type alias?

Reply:
The separate type is used to check if a function type is a constructor. 

If it's just an alias (`type ConstructorFunctionType = FunctionType`), then there is no way to distinguish the two.

If it's a type definition (`type ConstructorFunctionType FunctionType`), then that type is not implementing `Type`.

An alternative would be to e.g. just have a boolean field `IsConstructor` on `FunctionType`